pkg/cluster/k8sres.go

```diff
@@ -2380,3 +2384,18 @@ func ensurePath(file string, defaultDir string, defaultFile string) string {
 	}
 	return file
 }
+
+// mergeAnnotations merged annotations that entries in b override the ones in a.
+func mergeAnnotations(a, b map[string]string) map[string]string {
```

Comment:
When we will be on golang 1.18 with #2164, `maps.Copy` from the `golang.org/x/exp/maps` can be used instead.
Seems to this is too much resource and memory consumption implementation.
@FxKu wdyt about it?

Reply:
I read about `maps.Copy` yesterday. We would not need this function then and could do it in the code. Sounds good to me.

Reply:
Good call! Switched to use `maps.Copy` in https://github.com/zalando/postgres-operator/pull/2161/commits/3fb0b42f27c025ffe9e8088088ab0007262264f8